pkg/network/ebpf/c/protocols/http2/decoding-tls.h

```diff
@@ -157,25 +162,25 @@ static __always_inline __u8 tls_filter_relevant_headers(tls_dispatcher_arguments
             is_dynamic_table_update = (current_ch & 128) != 0;
             continue;
         }
-        // 224 is represented as 0b11100000, which is the OR operation for
-        // - indexed representation     (0b10000000)
-        // - literal representation     (0b01000000)
-        // - dynamic table size update  (0b00100000)
-        // Thus current_ch & 224 will be 0 only if the top 3 bits are 0, which means that the current byte is not
-        // representing any of the above.
-        if ((current_ch & 224) == 0) {
-            continue;
-        }
         // 32 is represented as 0b00100000, which is the scenario of dynamic table size update.
-        // From the previous condition we know that the top 3 bits are not 0, so if the top 3 bits are 001, then
-        // we have a dynamic table size update.
+        // If the top 3 bits are 001, then we have a dynamic table size update.
         is_dynamic_table_update = (current_ch & 224) == 32;
         if (is_dynamic_table_update) {
             continue;
         }
 
         is_indexed = (current_ch & 128) != 0;
-        max_bits = is_indexed ? MAX_7_BITS : MAX_6_BITS;
+        is_literal = (current_ch & 192) == 64;
+        // If all (is_indexed, is_literal, is_dynamic_table_update) are false, then we
+        // have a literal header field without indexing (prefix 0000) or literal header field never indexed (prefix 0001).
+
+        max_bits = MAX_4_BITS;
+        // If we're in an indexed header - the max bits are 7.
+        max_bits = is_indexed ? MAX_7_BITS : max_bits;
+        // else, if we're in a literal header - the max bits are 6.
+        max_bits = is_literal ? MAX_6_BITS : max_bits;
+        // otherwise, we're in literal header without indexing or literal header never indexed - and for both, the
```

Comment:
Can you maybe share as a comment link to the RFC which explains about this use case?